lib/type-format.js

```diff
@@ -0,0 +1,59 @@
+var XRegExp = require('xregexp');
+
+
+module.exports.formatType = function(type, nodeInfo) {
+  if (nodeInfo.kind === 'function') {
+    return {kind: "function", type: nodeInfo.typeInfo};
+  }
+
+  var allTypes = simplifyFunctions(simplifyObjects(type)).split("|");
+  var uniqModTypes = [];
+  for (var i = 0; i < allTypes.length; i++) {
+    if (uniqModTypes.indexOf(allTypes[i]) === -1) {
+      uniqModTypes.push(allTypes[i]);
```

Comment:
you can use `collectionsjs` functions (`collectionsjs` listed in package.json as a dependency)